Gems/ROS2/Code/Source/RobotImporter/RobotImporterWidgetUtils.cpp

```diff
@@ -0,0 +1,109 @@
+            if (result != QDialog::DialogCode::Accepted)
+            {
+                return AZStd::nullopt;
+            }
+
+            return importFileDialog.selectedFiles().first();
+        }
+
+        ExistingPrefabAction QueryUserForExistingPrefabAction(QWidget* parent = nullptr)
+        {
+            QMessageBox msgBox(parent);
+            msgBox.setWindowTitle(QObject::tr("Prefab file exists"));
+            msgBox.setText(QObject::tr("The prefab file already exists."));
+            msgBox.setInformativeText(QObject::tr("Do you want to overwrite it or save it with another file name?"));
+            msgBox.setStandardButtons(QMessageBox::Save | QMessageBox::Discard | QMessageBox::Cancel);
+            msgBox.setDefaultButton(QMessageBox::Discard);
+            msgBox.setButtonText(QMessageBox::Save, QObject::tr("Overwrite"));
+            msgBox.setButtonText(QMessageBox::Discard, QObject::tr("Save As..."));
+
+            switch (msgBox.exec())
+            {
+            case QMessageBox::Save:
+                return ExistingPrefabAction::Overwrite;
+            case QMessageBox::Discard:
+                return ExistingPrefabAction::CreateWithNewName;
+            default:
+                return ExistingPrefabAction::Cancel;
+            }
+        }
+    } // namespace Internal
+
+    AZStd::optional<AZStd::string> QueryUserForURDFPath(QWidget* parent)
+    {
+        const auto path = Internal::QueryUserForPath("Unified Robot Description Format (*.urdf)", QFileDialog::ExistingFiles);
+        if (!path)
+        {
+            return AZStd::nullopt;
+        }
+
+        if (path->isEmpty())
+        {
+            QMessageBox::critical(parent, QObject::tr("Empty path provided"), QObject::tr("No path was provided. Please try again"));
+            return QueryUserForURDFPath();
+        }
+
+        if (!QFile::exists(path.value()))
+        {
+            QMessageBox::critical(parent, QObject::tr("Does not exist"), QObject::tr("File does not exist. Please try again"));
+            return QueryUserForURDFPath();
+        }
+
+        return path->toUtf8().constData();
+    }
+
+    AZStd::optional<AZ::IO::Path> ValidatePrefabPathExistenceAndQueryUserForNewIfNecessary(const AZ::IO::Path& path, QWidget* parent)
+    {
+        if (!QFile::exists(path.c_str()))
+        {
+            return path;
+        }
+
+        switch (Internal::QueryUserForExistingPrefabAction(parent))
+        {
+        case ExistingPrefabAction::Cancel:
+            return AZStd::nullopt;
+        case ExistingPrefabAction::Overwrite:
+            return path;
+        case ExistingPrefabAction::CreateWithNewName:
+            // I am aware that similar functionality might be available by QFileDialog::setAcceptMode
+            // However, the prompt to confirm the overwrite showed up under the file selection dialog, which made a terrible UX
+            // TODO: It should be fixed at some point in the future
```

Comment:
A github issue should be made for this issue: https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#todo-comments

Reply:
Oh, it seems to be dead code. I will remove it after confirmation.